packages/reactive-element/src/css-tag.ts

```diff
@@ -189,6 +192,42 @@ export const adoptStyles = (
   }
 };
 
+const viewSheetCache = new WeakMap<
+  Window,
+  WeakMap<CSSStyleSheet, CSSStyleSheet>
+>();
+
+const sheetsForView = (view: Window, sheets: CSSStyleSheet[]) => {
+  let cache = viewSheetCache.get(view);
+  if (cache === undefined) {
+    viewSheetCache.set(view, (cache = new WeakMap()));
+  }
+
+  return sheets.map((sheet) => {
+    let viewSheet = cache.get(sheet);
+    if (viewSheet === undefined) {
+      if (
+        sheet instanceof
+        (view as unknown as {CSSStyleSheet: typeof CSSStyleSheet}).CSSStyleSheet
+      ) {
+        viewSheet = sheet;
+      } else {
+        const css = Array.from((sheet as CSSStyleSheet).cssRules).reduce(
```

Comment:
Use `cssResultFromStyleSheet` (defined in this module)

Reply:
Totally missed that. Thanks.